pandas/core/series.py

```diff
@@ -1045,13 +1047,13 @@ def _set_with(self, key, value):
             else:
                 key_type = lib.infer_dtype(key, skipna=False)
 
+            # Note: key_type == "boolean" should not occur because that
```

Comment:
can you add an assert for this

Reply:
did you want to add this here?

Reply:
hah, looks like i put the assertion with the corresponding comment in _get_with, not here in _set_with.  removed the assertion for symmetry.